pkg/accesscontrol/policy_rule_index.go

```diff
@@ -94,8 +94,22 @@ func addResourceAccess(accessSet *AccessSet, namespace string, rule rbacv1.Polic
 		for _, resource := range rule.Resources {
 			names := rule.ResourceNames
 			if len(names) == 0 {
-				names = []string{All}
+				// The first condition namespace != All is to determine if it is a RoleBinding.
+				// The second and third conditions are to check if the resource is for "namespaces" in core group.
+				// If names is empty, in the case of ClusterRoleBinding names is set to All.
+				// If names is empty, in the case of RoleBinding on the resource "namespaces" in core group, names is set to namespace
+				// in which the Rolebinding resides as kubernetes allows it and sets the requestInfo.Namespace to ResourceName.
+				// Ref: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go#L194
+				// In the case of Rolebinding, namespace is set to All in the case of resource "namespaces" in core group since namespace on
+				// the resource "namespaces" is not valid.
+				if namespace != All && resource == "namespaces" && group == "" {
+					names = []string{namespace}
+					namespace = All
```

Comment:
We can't override the `namespace` here without opening up for privesc when multiple resources are being requested. Otherwise, we may be changing the namespace scope to `*` for other Resources in this request.

Instead the logic needs to change so that the `accessSet.Add` for this specific edge case is handled here, without changing the `namespace` var for everybody else.

Reply:
Agreed.